AbletonMCP_Remote_Script/handlers/devices.py

```
@@ -0,0 +1,1358 @@
+    track_type="track", value_display=None, ctrl=None
+):
+    """Set a device parameter by name on any track type.
+
+    value_display: optional display string (e.g. '1/4') for quantized params.
+    If provided, overrides the numeric value.
+    """
+    try:
+        track = resolve_track(song, track_index, track_type)
+        device_list = list(track.devices)
+        if device_index < 0 or device_index >= len(device_list):
+            raise IndexError("Device index out of range")
```

Ruff (0.15.0)

[warning] Abstract `raise` to an inner function

(TRY301)